### PR TITLE
Allow to manage plugins.

### PR DESCRIPTION
**Pull request by ypid**
I wanted to wait for feedback on https://github.com/ansible/ansible/issues/12710 but this might take a while.

### CODE REVIEW

**Comment by carlalexander** [comment]
Yeah, let's not wait on that Ansible PR. I'll be looking at this one next. The only thing that stands out is seperate option for `latest`. Why not use another state there as well?

**Comment by ypid** [comment]
To allow flexibility. That is the way most core modules do it …

**Comment by carlalexander** [comment]
Aren't packages and such all `absent`, `present` and `latest`?. I figure the behaviour is closer to that of a package.

**Comment by carlalexander** [comment]
Oh wait, I see the issue. We can't say `active` and `latest`. But in that scenarion, doesn't it make more sense to have `active` as a boolean. Then we use `state` with the other three.

**Comment by ypid** [comment]
You are right. I think that makes more sense.

**Comment by carlalexander** [comment]
Cool! Glad I'm not crazy 😄

**Comment by ypid** [comment]
I just tested it again (after Ansible update from 1.9.3-1 to 1.9.4-1) and it seems that my implementation requires jinja2 2.8+ version of jinja2 ([issue](https://github.com/ansible/ansible/issues/8836)). Other implementations are not so clean (using `groupby`). Maybe we should wait a while before merging.

**Comment by ypid** [comment]
Updated. `enabled` is used in Ansible 2 `service` module.

**Comment by carlalexander** [comment]
Ok, sounds good. I'll look into it. Another thing I'm not sure about. Default plugins. Maybe that should be in the documentation vs built in. Thoughts?

**Comment by ypid** [comment]
I think 'ricg-responsive-images', 'redis-cache' and 'akismet' are a good start. If someone does not want them they can still disable them somehow.

**Comment by carlalexander** [comment]
I'm just not sure if it's our job to make these recommendations beyond the documentation. I had the Redis object cache installed by default before and took it out for that reason.

**Comment by ypid** [comment]
Sure, when you tried that already. My experience with maintaining Wordpress instances is limited. Updated PR.

**Comment by carlalexander** [comment]
I'm looking at the jinja issue vs your code. We could use `wp plugin list --status=active` to get the active list of plugins and skip `selectattr`. It's an extra steps for sure. 

Is there no way to create the dictionary without the two steps?

**Comment by ypid** [comment]
A little less Ansible_ish_ but works as well :smile: Updated.

File: roles/wordpress/tasks/manage_plugins.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+- name: Get list of active WordPress plugins
+  command: 'wp plugin list --format=json --status=active'
+  sudo_user: '{{ wordpress_user }}'
+  args:
+    chdir: '{{ wordpress_root }}'
+  changed_when: False
+  always_run: True
+  register: wordpress_plugin_list_active_cmd
+
+- name: Convert list of installed WordPress plugins from JSON
```
Comment (carlalexander) [inline roles/wordpress/tasks/manage_plugins.yml]: Any reason why this is its own step? I tried combining the two like `wordpress_plugin_list_installed_cmd.stdout|from_json|map(attribute="name")|list`. It seemed to work fine.

Comment (ypid) [inline roles/wordpress/tasks/manage_plugins.yml]: Updated.

"Two tasks provide no benefit with the current implementation but
had the advantage of avoiding redundancy when one `command` task could
be used to generate the two lists."

**Comment by carlalexander** [comment]
So still running some tests, I'm getting an error when activating `redis-object-cache`. I think it's more of a bug with cli or the plugin. Otherwise, I think it might be good. I'll run a few more tests tomorrow (I hope).

**Comment by carlalexander** [comment]
Ok, I'm an idiot. The right plugin is `redis-cache`. Everything looks good. Would you mind squashing all the commits together? Otherwise, I can take care of it.

**Comment by ypid** [comment]
Done.

**Comment by carlalexander** [comment]
Thanks! :ok_hand:

**Comment by carlalexander** [comment]
Alright, I just merged your commit. Thanks again, this is an awesome change! I'm leaving this open with a note to write some wiki documentation for it.

**Comment by carlalexander** [comment]
[Wiki page](https://github.com/carlalexander/debops-wordpress/wiki/How-to-manage-WordPress-plugins) created. We're good now!

**Pull request closed without merging**